test/cli/packages_cmd_test.go

```diff
@@ -137,3 +137,75 @@ func TestPackagesCmdFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestRegistryAuth(t *testing.T) {
+	tests := []struct {
+		name       string
+		args       []string
+		env        map[string]string
+		assertions []traitAssertion
+	}{
+		{
+			name: "fallback to keychain",
+			args: []string{"packages", "-vv", "registry:localhost:5000/something:latest"},
+			assertions: []traitAssertion{
+				assertInOutput("source=OciRegistry"),
+				assertInOutput("localhost:5000/something:latest"),
+				assertInOutput("no registry credentials configured, using the default keychain"),
+			},
+		},
+		{
+			name: "use creds",
+			args: []string{"packages", "-vv", "registry:localhost:5000/something:latest"},
+			env: map[string]string{
+				"SYFT_REGISTRY_AUTH_AUTHORITY": "localhost:5000",
+				"SYFT_REGISTRY_AUTH_USERNAME":  "username",
+				"SYFT_REGISTRY_AUTH_PASSWORD":  "password",
+			},
+			assertions: []traitAssertion{
+				assertInOutput("source=OciRegistry"),
+				assertInOutput("localhost:5000/something:latest"),
+				assertInOutput(`using registry credentials for "localhost:5000"`),
+			},
+		},
+		{
+			name: "use token",
+			args: []string{"packages", "-vv", "registry:localhost:5000/something:latest"},
+			env: map[string]string{
+				"SYFT_REGISTRY_AUTH_AUTHORITY": "localhost:5000",
+				"SYFT_REGISTRY_AUTH_TOKEN":     "token",
+			},
+			assertions: []traitAssertion{
+				assertInOutput("source=OciRegistry"),
+				assertInOutput("localhost:5000/something:latest"),
+				assertInOutput(`using registry token for "localhost:5000"`),
+			},
+		},
+		{
+			name: "not enough info fallsback to keychain",
+			args: []string{"packages", "-vv", "registry:localhost:5000/something:latest"},
+			env: map[string]string{
+				"SYFT_REGISTRY_AUTH_AUTHORITY": "localhost:5000",
+			},
+			assertions: []traitAssertion{
+				assertInOutput("source=OciRegistry"),
+				assertInOutput("localhost:5000/something:latest"),
+				assertInOutput(`no registry credentials configured, using the default keychain`),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cmd, stdout, stderr := runSyftCommand(t, test.env, test.args...)
+			for _, traitFn := range test.assertions {
+				traitFn(t, stdout, stderr, cmd.ProcessState.ExitCode())
```

Comment:
nit: It might be more clear when reading line 202 if the individual slice item was named something like `assertFn` (or something that includes "assert") — just since it's a recognizable/idiomatic part of a test body